lib/clAdapters/jira/index.js

```diff
@@ -0,0 +1,66 @@
+import Adapter from '../base/Adapter';
+import url from 'url';
+import request from 'request';
+
+export default class JiraAdapter extends Adapter {
+  constructor() {
+    super();
+    this.apiVersion = 2;
+  }
+
+  init() {
+  }
+}
+
+JiraAdapter.prototype.makeRequest = function(path) {
+  const uri = url.format({
+    protocol: this.credentials.protocol || 'https',
+    hostname: this.credentials.host,
+    port: this.credentials.port,
+    pathname: 'rest/api/' + this.apiVersion + '/' + path
+  });
+
+  const authorizationString = new Buffer(this.credentials.username + ':' + this.credentials.password).toString('base64');
+
+  const options = {
+    uri: uri,
+    method: 'GET',
+    headers: {
+      'Authorization': 'Basic ' + authorizationString
+    }
+  };
+
+  return new Promise((resolve) => {
+    request(options, (error, response, body) => {
+      let errorMessage = null;
+      let success = response && response.statusCode < 400;
```

Comment:
Not really, it's an approximation. There are a lot of codes in use and generally the ones >= 400 looked to be errors. https://developer.atlassian.com/confdev/development-resources/confluence-developer-faq/http-response-code-definitions

Definitely open to other suggestions on how to handle this